dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -155,12 +160,17 @@ int FramesDecoderGpu::ProcessPictureDecode(void *user_data, CUVIDPICPARAMS *pict
     Width(),
     Height(),
     stream_);
+  // TODO(awolant): Alterantive is to copy the data to a buffer
+  // and then process it on the stream. Check, if this is faster, when
+  // the benchamrk is ready.
+  CUDA_CALL(cudaStreamSynchronize(stream_));
```

Comment:
Even though we pass `stream_` do NVDEC it turns out it is necessary to synchronize the stream before we unmap the frame. 
It is not a big problem as we do not do anything else on it and synchronize it immediately after `Run`.
When we have proper benchamrk (DALI-2594) we can test the alternative approach and pick the better one.